### PR TITLE
arch/armv7-m/assert: show stackdump regardless of binary manager

### PR DESCRIPTION
**Pull request by sunghan-chang**
CONFIG_ARCH_STACKDUMP controls stackdump feature. Regardless of
binary manager enabled, it should be working with that config.
This commit makes stackdump independent from binary manager.

And tidyup of assert functions.

Signed-off-by: sunghan-chang <sh924.chang@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [7e9e74484c157c8fd9fd7208deaf72a14126d727] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7e9e74484c157c8fd9fd7208deaf72a14126d727] - Code Rule Check OK.

File: os/arch/arm/src/armv7-m/up_assert.c [hunk 1]
```diff
@@ -408,57 +408,12 @@ static void up_dumpstate(void)
...
-		if (current_regs) {
-			tcb = sched_self();
-			current_regs = tcb->xcp.regs;
-		}
-	}
-#else
-
 #ifndef CONFIG_BOARD_ASSERT_SYSTEM_BLOCK
 	/* Are we in an interrupt handler or the idle task? */
+
```
Comment (NilabhShant) [inline os/arch/arm/src/armv7-m/up_assert.c]: Please remove this extra line.

File: os/arch/arm/src/armv7-m/up_assert.c [hunk 2]
```diff
@@ -519,21 +507,38 @@ void up_assert(const uint8_t *filename, int lineno)
...
+	if (is_kernel_assert == false) {
+		/* recovery user assert through binary manager */
+
+		recovery_user_assert();
+	} else
 #endif
-	_up_assert(EXIT_FAILURE);
+	{
+		/* treat kernel assert */
+
```
Comment (NilabhShant) [inline os/arch/arm/src/armv7-m/up_assert.c]: same as above.

File: os/arch/arm/src/armv7-m/up_assert.c (hunk 1 above)
Comment (sunghan-chang) [reply os/arch/arm/src/armv7-m/up_assert.c]: I just follow Nuttx style. Nuttx gives an empty line after comment.

**Comment by seinfra** [comment]
Target : [48432e496499dafe1911f3900730a7806478360f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [48432e496499dafe1911f3900730a7806478360f] - Code Rule Check OK.

File: os/arch/arm/src/armv7-m/up_assert.c (hunk 1 above)
Comment (NilabhShant) [reply os/arch/arm/src/armv7-m/up_assert.c]: Ok, As it was not there previously so I leave comment on it.

**Comment by seinfra** [comment]
Target : [8ecaca105d8da8e057b5ca6bc2e13157826362ab] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8ecaca105d8da8e057b5ca6bc2e13157826362ab] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4b917579f4ba2f5b0e1f4f5e217ffb944af33f78] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4b917579f4ba2f5b0e1f4f5e217ffb944af33f78] - Code Rule Check (C++) OK.

**Comment by hasw7569** [comment]
It looks good to me

File: os/arch/arm/src/armv7-m/up_assert.c [hunk 3]
```diff
@@ -470,13 +435,46 @@ static void _up_assert(int errorcode)
...
+	request_msg.cmd = BINMGR_FAULT;
+	request_msg.requester_pid = getpid();
+	if (current_regs) {
+		tcb = sched_self();
+		sched_removereadytorun(tcb);
+#if CONFIG_RR_INTERVAL > 0
+		tcb->timeslice = 0;
 #endif
+		tcb->sched_priority = SCHED_PRIORITY_MIN;
```
Comment (btheosam) [inline os/arch/arm/src/armv7-m/up_assert.c]: This block of code has to be looked into later.

**Pull request merged**